internal/v1/db/client/stats.go

```diff
@@ -46,6 +46,150 @@ func (db *V1Database) GetOrCreateStatsLock(
 	return &result, nil
 }
 
+// UpdateLegacyOverallStats updates the legacy overall stats for babylon staking
+// This is a temporary method run the the ad-hoc script to update stats in a
+// single run
+func (v1dbclient *V1Database) UpdateLegacyOverallStats(
+	ctx context.Context,
+) (*v1dbmodel.OverallStatsDocument, error) {
+	var totalDelegations, totalTvl, activeDelegations, activeTvl uint64
+
+	overallStatsClient := v1dbclient.Client.Database(
+		v1dbclient.DbName,
+	).Collection(dbmodel.V1OverallStatsCollection)
+	stakerStatsClient := v1dbclient.Client.Database(
+		v1dbclient.DbName,
+	).Collection(dbmodel.V1StakerStatsCollection)
+	delegationClient := v1dbclient.Client.Database(
+		v1dbclient.DbName,
+	).Collection(dbmodel.V1DelegationCollection)
+
+	// Start a session for transaction
+	session, sessionErr := v1dbclient.Client.StartSession()
+	if sessionErr != nil {
+		return nil, sessionErr
+	}
+	defer session.EndSession(ctx)
+
+	// Define the work to be done in the transaction
+	transactionWork := func(sessCtx mongo.SessionContext) (interface{}, error) {
+		// Calculate the total stakers
+		// Count total number of stakers in the staker stats collection
+		totalStakersCount, err := stakerStatsClient.CountDocuments(
+			sessCtx,
+			bson.M{},
+		)
+		if err != nil {
+			return nil, err
+		}
+		totalStakers := uint64(totalStakersCount)
+
+		// Count total number of delegations and sum their staking values
+		aggregateResult, err := delegationClient.Aggregate(
+			sessCtx,
+			[]bson.M{
+				{
+					"$facet": bson.M{
+						"total": []bson.M{
+							{
+								"$match": bson.M{
+									"is_overflow": false,
+								},
+							},
+							{
+								"$group": bson.M{
+									"_id":        nil,
+									"count":      bson.M{"$sum": 1},
+									"totalValue": bson.M{"$sum": "$staking_value"},
+								},
+							},
+						},
+						"active": []bson.M{
+							{
+								"$match": bson.M{
+									"state":       "active",
+									"is_overflow": false,
+								},
+							},
+							{
+								"$group": bson.M{
+									"_id":        nil,
+									"count":      bson.M{"$sum": 1},
+									"totalValue": bson.M{"$sum": "$staking_value"},
+								},
+							},
+						},
+					},
+				},
+			},
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		type groupResult struct {
+			Count      int64 `bson:"count"`
+			TotalValue int64 `bson:"totalValue"`
+		}
+
+		var results struct {
+			Total  []groupResult `bson:"total"`
+			Active []groupResult `bson:"active"`
```

Comment:
Is it true that it's either empty slice or slice with 1 value? If this is the case maybe rewrite it with `*groupResult`?

Reply:
no worries, let's keep it as is since this script will be launched just a few times max